src/rpg_database/character_sheet/LevelAndXPCrossDependencyCheck.java

```diff
@@ -0,0 +1,98 @@
+package rpg_database.character_sheet;
+
+import java.security.InvalidParameterException;
+import java.util.HashMap;
+
+import rpg_database.character_sheet.exceptions.CrossDependencyException;
+import rpg_database.character_sheet.interfaces.MultipleFieldsGetterSetter;
+
+public class LevelAndXPCrossDependencyCheck implements MultipleFieldsGetterSetter<LevelAndXPCrossDependencyCheck, Integer> {
+
+	private int xp;
+	private int level;
+	private final int maxLevel = 22;
+	private final int maxXp = 70000;
+
+	private final HashMap<Integer, Integer> xpNeedingForLevelMap = fillUpWithLevelXpPairs();
+
+	public LevelAndXPCrossDependencyCheck() {
+		xp = 0;
+		level = 1;
+	}
+
+	public int GetXP() {
+		return xp;
+	}
+
+	@Override
+	public Class<LevelAndXPCrossDependencyCheck> getImplementingClass() {
+		return LevelAndXPCrossDependencyCheck.class;
+	}
+
+	@Override
+	public Class<Integer> getDataTypeClass() {
+		return Integer.class;
+	}
+
+	@Override
+	public void setSelfValueByField(Fields field, Integer value) {
+		if (field == Fields.XP) {
+			if (value > maxXp)
+				throw new CrossDependencyException("The xp gap is: " + maxXp + "!");
+			if (level == 1 || xpNeedingForLevelMap.get(level) <= value) {
+				xp = value;
+				while (xpNeedingForLevelMap.get(level) < xp && xpNeedingForLevelMap.get(level + 1) <= xp) {
+					level++;
+				}
+			} else {
+				throw new CrossDependencyException("Your level is higher than the given xp!");
+			}
+		} else if (field == Fields.LEVEL) {
+			if (value > maxLevel)
+				throw new CrossDependencyException("The level gap is: " + maxLevel + "!");
+			if (xp == 0 || xpNeedingForLevelMap.get(value) <= xp) {
+				level = value;
+				xp = xpNeedingForLevelMap.get(value);
+			} else
+				throw new CrossDependencyException("You don't have enough xp for this level!");
+		} else {
+			throw new InvalidParameterException("It's not a valid Field type!");
+		}
+	}
+
+	@Override
+	public Integer getStoredValueByField(Fields field) {
+		if (field == Fields.XP)
+			return xp;
+		else if (field == Fields.LEVEL)
+			return level;
+		throw new InvalidParameterException("It's not a valid Field type!");
+	}
+
+	private HashMap<Integer, Integer> fillUpWithLevelXpPairs() {

```

Comment:
Ennek nem olyan bonyolult ám a matekja. Ezt miért nem oldod meg a kóddal?
```java
if(level <= 1)
    xp = 0;
if( level < 7 )
    xp = (level-1)*2000;
else if(level < 13)
    xp = 10000 + (level -7)*3000;
else
    xp = 28000 + (level - 13)*4000;
return xp;
``` 
ez első ránézésre amit írnék, de nem teszteltem, meg lehet hogy még ezen is lehet kicsit finomítani....

Reply:
Itt egy másik:
```java
int getXPRequiredForLevel(int level){
    int xp = 0;
    int[] levelCaps = {13, 7, 1};
    int increment = 4000;
    for(int cap: levelCaps){
        increment-=1000;
        while(level - cap > 1){
            xp += increment;
            level--;
        }
    }
    return xp;
}
```
ezt sem teszteltem de talán faszább mint az előző.... aztán ezen is lehet sokat finomítani, kirakni a levelCaps-et statikus final konstans kutyafaszommá meg hadd ne soroljam